### ISSUE TITLE
Handle lockscreen separately?

### ISSUE THREAD

**Issue by quezak** [hhannine/superpaper#31]
First of all, congrats for making this awesome package! I was looking for a solution to just properly span the wallpaper over my not-so-simple 3 monitor setup, never expected to find PPI corrections and even including bezels into the calculations! :smiley: 

It looks really good after tweaking the offsets & bezel sizes [1], but there is one issue -- when showing the lock screen [2], only one of the monitors is active, and it shows what would be the wallpaper for all 3 monitors (blue-green solid color background added to clearly see the wallpaper boundaries). Would it be possible to setup a separate wallpaper for the lockscreen, or make the lockscreen use the config for just the center monitor?

[1] 
![IMG_20200224_143551](https://user-images.githubusercontent.com/666206/75158017-35fa2a00-570d-11ea-88ba-9000c96c1fd2.jpg)
(I just saw the glitch in the bottom-left part of the wallpaper is not superpaper's fault, just the wallpaper file is broken)

[2]
![IMG_20200224_143604](https://user-images.githubusercontent.com/666206/75158031-3abede00-570d-11ea-8eb0-95ac2f0706c5.jpg)

Config:
* arch linux, fully updated yesterday
* superpaper 1.2.0
* mate 1.22.x (with x in 0,1,2,4 depending on the package)
* python 3.8
* display setup:
   * left 1920x1200 24" portait
   * center 2560x1600 27" landscape
   * right (laptop screen) 1920x1080 15.6" portait
* anything else I should add?

**Comment by quezak** [comment]
alternatively, maybe there is a way to just activate all monitors for the lockscreen? I can't seem any config for the MATE lockscreen appearance, but I'll look more later

**Comment by hhannine** [comment]
Thanks, glad to hear it's mostly working well! :smile: 

Some quick thoughts:
- Implementing this would be Desktop Environment / lock screen provider specific. First step for you with Mate would be to figure out if Mate will handle multiple displays. If it doesn't, it would be  possible crop the primary monitor piece from the wallpaper but this would need new logic since superpaper is currently unaware what is the primary monitor on which the lock screen would be displayed. And it would be preferable to have the lock screen image be separate from the desktop wallpaper image if this is the solution one would have to implement, since if they're the one and the same, to change the cropped image as the lock screen wallpaper, superpaper would need to change the desktop wallpaper to be the cropped one. Superpaper would have to monitor lock screen state which would be quite undesirable.

- On my RHEL 7 Cinnamon 3.6 machine at work the lock screen is using both of my monitors and is showing the spanned wallpaper on them as it does on the desktop so Cinnamon already works as desired.

- On my Manjaro KDE machine at home both monitors are active in lock screen and a default wallpaper is shown on each monitor, a copy of the same image on each display. With some brief googling it sounds a bit like it would be possible to control the lock screen wallpaper in a similar manner as it is done on the desktop. This will need some work and figuring out the API keys to access the lock screen wallpaper, which specifically I could not now find.

- All other DEs / lock screens that do not work now would need to be reported and researched how their lock screen wallpapers are managed. It is dependent on the lock screen provider to support multiple monitors, i.e. spanning a wallpaper across all monitors in the lock screen or alternatively allow setting a separate image for each monitor like KDE, superpaper has no way to extend that.

**Comment by quezak** [comment]
Yeah, that's what I've suspected, it's too dependent on the particular DE config. I've discovered it's even more complicated in my case, as the lockscreen doesn't always show on the center monitor, but rather on the one where the cursor located :sweat_smile: I'll try to look deeper into lockscreen config to find a workaround, lockscreens aren't visible too long so it's not that important.

**Comment by hhannine** [comment]
Yeah. One workaround could be to set the lock screen to use a different image from the desktop image so you wouldn't need to look at the ugly composite image. If there isn't an option to enable all of the monitors in the lock screen.

**Comment by hhannine** [comment]
A bit bad news for the future of this feature. I had one idea of how to try to implement muli-monitor wallpaper in the KDE lockscreen and unfortunately it does not work, which matches the claims I found online that it is not possible.
( I tried to replace the line
```
desktop.currentConfigGroup = Array("Wallpaper", "org.kde.image", "General");
```
with
```
desktop.currentConfigGroup = Array("Greeter", "Wallpaper", "org.kde.image", "General");
```
based on the answer here: https://forum.manjaro.org/t/set-lockscreen-background-in-kde-from-terminal/66539 )

Second bad news is that on Windows 10 the lockscreen wallpaper is for whatever reason stored in a binary file that has no API to edit, and trying to edit it yourself can cause the whole login screen to break. Pretty bad. I read that there are some third party tools to change the wallpaper but I'm not too excited. Maybe some optional plug-in feature at best, but probably no support for  Windows either.

However, I did find out how to change the wallpaper on the 'gsettings' family of DEs: https://askubuntu.com/a/1181775, though this does not provide any control over the multi-monitor aspect of the lock screen.
